### PR TITLE
rustbuild: allow overriding list of LLVM targets to build

### PR DESCRIPTION
**Pull request by xen0n**
Here we go!

r? @alexcrichton
cc @Yamakaky

### CODE REVIEW

File: src/bootstrap/native.rs [hunk 1]
```diff
@@ -75,13 +75,18 @@ pub fn llvm(build: &Build, target: &str) {
         (true, true) => "RelWithDebInfo",
     };
 
+    // NOTE: remember to also update `config.toml.example` when changing the defaults!
```
Comment (rkruppe) [inline src/bootstrap/native.rs]: :+1:

**Comment by Yamakaky** [comment]
Cool!

**Comment by xen0n** [comment]
Wait there's the problem of stale cache when the target list has been changed. Most likely we'd like to revamp the current stamping logic to include the targets also, maybe ditching the date part altogether. I'll come up with something like that later in the day.

**Comment by alexcrichton** [comment]
@bors: r+

Looks good to me! I don't think we have to worry too much about stale issues here, presumably you know what you're doing if you're tweaking this configuration.

**Comment by Yamakaky** [comment]
Auto-rebuild would be cool though.

**Pull request merged**